### DISCUSSION TITLE
How do you save workflows in a Kubernetes HA deployment?

### DISCUSSION THREAD

**Discussion by sanjayh94** [StackStorm/st2#5634] [Q&A]
I'm new to the K8s HA deployment and have a deployed stackstorm 3.6 in a GKE cluster. I get an Internal Server Error 500 when I try to save an Orquestra workflow from the web UI. Is there an issue with this?

I'm installing packs using the pack image method. Do I need separate storage to save workflows? or Can I bake them in images just like the packs?

**Comment by arm4b** [comment]
Yes, you need to use `Method 2: Shared Volumes` for managing the pack content.
Shared Read-Write-Many storage is required for any functionality to work that saves data to the file system from the st2 cluster (similar to `st2 pack install`):

https://github.com/StackStorm/stackstorm-k8s#method-2-shared-volumes

**Reply by flyte** [reply]
This information would be very helpful in the readme that you linked to.

It ought to be made clear that the whole workflow editor feature simply will not work unless you set up the 'Shared Volumes' configuration.

**Comment by kylehartigan-idea11** [comment]
Is it possible to use a combination of both methods?

for example, all my custom packs are built using the docker method (which I do believe is better for pack codebase changes and control etc) and then have a persistent volume mapped to /opt/stackstorm/packs/default that will allow users to build their own workflows there?
